### DISCUSSION TITLE
Code action duplicate entries

### DISCUSSION THREAD

**Discussion by davidsoles** [mfussenegger/nvim-jdtls#293] [Q&A] [answered]
Hi,
Sometimes code actions menu duplicates the options. Is there anything I can do about it?
![image](https://user-images.githubusercontent.com/984410/178399166-77baca19-6fa2-4f71-b6e7-252fe552e425.png)
Thanks.

**Comment by mfussenegger** [comment]
Do you by chance have multiple LSP clients running? (Check the `vim.lsp.get_active_clients()` output). 

The code actions are otherwise generated by eclipse.jdt.ls and neovim only displays them. This would be a server bug, but I haven't encountered this.

**Comment by abenz1267** [comment]
I get this all the time as well. If you actually take a look at the jdtls response for code actions, you'll see that some code actions are duplicated with different kinds.

**Comment by mfussenegger** [comment, marked as answer]
Reported this upstream: https://github.com/eclipse/eclipse.jdt.ls/issues/2177
